test/celeritas/field/LinearPropagator.test.cc

```diff
@@ -59,9 +59,10 @@ class LinearPropagatorTestBase : public GlobalGeoTestBase
     }
 
   protected:
-    SPConstParticle build_particle() override { CELER_ASSERT_UNREACHABLE(); }
-    SPConstCutoff   build_cutoff() override { CELER_ASSERT_UNREACHABLE(); }
-    SPConstPhysics  build_physics() override { CELER_ASSERT_UNREACHABLE(); }
+    SPConstParticle  build_particle() override { CELER_ASSERT_UNREACHABLE(); }
+    SPConstCutoff    build_cutoff() override { CELER_ASSERT_UNREACHABLE(); }
+    SPConstPhysics   build_physics() override { CELER_ASSERT_UNREACHABLE(); }
+    SPConstTrackInit build_init() override { CELER_ASSERT_UNREACHABLE(); }
```

Comment:
Doesn't have to be part of this MR, but I think we should have a mixin class that provides "unreachable" implementations:
```c++
class OnlyGeoTestBase : virtual public GlobalTestBase
{
  public:
    SPConstParticle build_particle() override { CELER_ASSERT_UNREACHABLE(); }
    SPConstCutoff   build_cutoff() override { CELER_ASSERT_UNREACHABLE(); }
    SPConstPhysics  build_physics() override { CELER_ASSERT_UNREACHABLE(); }
    SPConstAction   build_along_step() override { CELER_ASSERT_UNREACHABLE(); }
    SPConstMaterial build_material() override { CELER_ASSERT_UNREACHABLE(); }
    SPConstGeoMaterial build_geomaterial() override
    {
        CELER_ASSERT_UNREACHABLE();
    }
};
```
and then we'd have
```public
class LinearPropagatorTestBase : public GlobalGeoTestBase, public OnlyGeoTestBase
{ /* */ };
```